darts/models/forecasting/lgbm.py

```diff
@@ -195,11 +257,66 @@ def fit(
             past_covariates=past_covariates,
             future_covariates=future_covariates,
             max_samples_per_ts=max_samples_per_ts,
+            categorical_past_covariates=categorical_past_covariates,
+            categorical_future_covariates=categorical_future_covariates,
+            categorical_static_covariates=categorical_static_covariates,
             **kwargs,
         )
 
         return self
 
+    def _fit_model(
+        self,
+        target_series,
+        past_covariates,
+        future_covariates,
+        max_samples_per_ts,
+        **kwargs,
+    ):
+        """
+        Custom fit function for the LightGBM model; adding logic to let the model handle categorical features
+        directly.
+        """
+
+        training_samples, training_labels = self._create_lagged_data(
+            target_series,
+            past_covariates,
+            future_covariates,
+            max_samples_per_ts,
+        )
+
+        # Check if there are covariates that should be treated as categorical
+        categorical_past_covariates = kwargs.pop("categorical_past_covariates", None)
```

Comment:
Assume we move the categorical variable definition to model creation. We can already process the values and store the `categorical_covariates` in `self`.

Then we have access to these variables already and we can simply call your `_get_categorical_features()` on `series`, `past_covariates`, and `future_covariates.